x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/transport/actions/TransportWatcherStatsActionTests.java

```diff
@@ -62,6 +66,7 @@ public void setupTransportAction() {
         ClusterState clusterState = mock(ClusterState.class);
         when(clusterState.getMetadata()).thenReturn(Metadata.EMPTY_METADATA);
         when(clusterService.state()).thenReturn(clusterState);
+        when(clusterState.metadata()).thenReturn(Metadata.builder().put(ProjectMetadata.builder(projectId).build()).build());
```

Comment:
Can we avoid returning two different metadata objects to avoid confusion? Ideally, we'd align the production code to only use one of the two getter methods (`metadata()` being the preference), but if that results in a very large number of changes, I'm also ok with leaving that part.